estimator/util.py

```diff
@@ -366,28 +391,20 @@ def batch_estimate(params, algorithm, jobs=1, log_level=0, catch_exceptions=True
         >>> _ = batch_estimate(Kyber512, [LWE.primal_usvp, LWE.primal_bdd], jobs=2)
 
     """
-    from .lwe_parameters import LWEParameters
 
     if isinstance(params, LWEParameters):
         params = (params,)
     if not hasattr(algorithm, "__iter__"):
         algorithm = (algorithm,)
-
-    tasks = [(partial(f, **kwds), x, log_level, f_name(f), catch_exceptions) for x in params for f in algorithm]
+    tasks = [
+        Task(partial(f, **kwds), x, log_level, f_name(f), catch_exceptions)
+        for f, x in it.product(algorithm, params)
+    ]
 
     if jobs == 1:
-        res = {
-            (f_repr, x): _batch_estimatef(f, x, lvl, f_repr, catch_exceptions)
-            for f, x, lvl, f_repr, catch_exceptions in tasks
-        }
+        results = [_batch_estimatef(*task) for task in tasks]
     else:
-        pool = Pool(jobs)
-        res = pool.starmap(_batch_estimatef, tasks)
-        res = {(f_repr, x): res[i] for i, (_, x, _, f_repr, _) in enumerate(tasks)}
-
-    ret = {x: {} for _, x in res.keys()}
-    for (f, x), v in res.items():
-        if v is not None:
-            ret[x][f] = v
+        with Pool(min(jobs, cpu_count())) as pool:
```

Comment:
Why put the limit there?

Reply:
Why have the limit or why specifically put it there and not elsewhere?
For the former, if you have more job processes than cpus that can run them, your throughput will go down. For the latter, it seems like the place to naturally put it; where the `Pool` is created. Thinking about it though, maybe a writting to the log in case that the number of processes are clipped to the max number may be useful.

Reply:
Ah, sorry. I meant why impose a limit at all. I'd say if a user wants to use 200 jobs on a system that reports 8 threads, let them?

Reply:
Ok. Change reverted :)